Compiling fails with boost 1.74

Bug Report
==========
It seems that there are some changes in the boost library. 
Cmake is warning about this. So it's more a todo then a bug. 

freeorion/GG/src/Texture.cpp:244:86:   required from here
/usr/include/boost/gil/extension/dynamic_image/any_image.hpp:53:52: error: 'struct boost::mp11::mp_list<boost::gil::image<boost::gil::pixel<unsigned char, boost::gil::layout<boost::mp11::mp_list<boost::gil::gray_color_t>, boost::mp11::mp_list<std::integral_constant<int, 0> > > >, false, std::allocator<unsigned char> >, boost::gil::image<boost::gil::pixel<unsigned char, boost::gil::layout<boost::mp11::mp_list<boost::gil::gray_color_t, boost::gil::alpha_t>, boost::mp11::mp_list<std::integral_constant<int, 0>, std::integral_constant<int, 1> > > >, false, std::allocator<unsigned char> >, boost::gil::image<boost::gil::pixel<unsigned char, boost::gil::layout<boost::mp11::mp_list<boost::gil::red_t, boost::gil::green_t, boost::gil::blue_t>, boost::mp11::mp_list<std::integral_constant<int, 0>, std::integral_constant<int, 1>, std::integral_constant<int, 2> > > >, false, std::allocator<unsigned char> >, boost::gil::image<boost::gil::pixel<unsigned char, boost::gil::layout<boost::mp11::mp_list<boost::gil::red_t, boost::gil::green_t, boost::gil::blue_t, boost::gil::alpha_t>, boost::mp11::mp_list<std::integral_constant<int, 0>, std::integral_constant<int, 1>, std::integral_constant<int, 2>, std::integral_constant<int, 3> > > >, false, std::allocator<unsigned char> > >' has no member named 'recreate'
   53 |     result_type operator()(Image& img) const { img.recreate(_dimensions,_alignment); }


Environment
-----------
Linux: Gentoo ~amd64
gcc (Gentoo 10.2.0-r1 p2) 10.2.0
Python 3.7.9
boost 1.74

* **FreeOrion Version**:
Building v0.4.10+ build 2020-09-16.4ce1758

* **Operating System**:
Gentoo Linux
* Compiled from source


Description
-----------
compiling failed 

Expected Result
---------------
compile freeorion

Steps to reproduce
------------------
mkdir build 
cd build 
cmake ..
make 

Kind regards Bernd Wollny

--- a/GG/src/Texture.cpp
+++ b/GG/src/Texture.cpp
@@ -24,7 +24,9 @@
 #  include <boost/gil/extension/io/png.hpp>
 # endif
 #endif
-#if BOOST_VERSION >= 107000
+#if BOOST_VERSION >= 107400
+#include <boost/variant2/variant.hpp>
+#elif BOOST_VERSION >= 107000
 #include <boost/variant/get.hpp>
 #endif
 #include <GG/GLClientAndServerBuffer.h>
@@ -216,18 +218,24 @@ void Texture::Load(const boost::filesystem::path& path, bool mipmap/* = false*/)
     static_assert(sizeof(gil::rgb8_pixel_t) == 3, "rgb8 pixel type does not match expected type size");
     static_assert(sizeof(gil::rgba8_pixel_t) == 4, "rgba8 pixel type does not match expected type size");
 
-#ifdef BOOST_GIL_USES_MP11
-    typedef boost::mp11::mp_list<
+#if BOOST_VERSION >= 107400
+    typedef gil::any_image<gil::gray8_image_t,
+        gil::gray_alpha8_image_t,
+        gil::rgb8_image_t,
+        gil::rgba8_image_t> ImageType;
 #else
+# ifdef BOOST_GIL_USES_MP11
+    typedef boost::mp11::mp_list<
+# else
     typedef boost::mpl::vector4<
-#endif
+# endif
         gil::gray8_image_t,
         gil::gray_alpha8_image_t,
         gil::rgb8_image_t,
         gil::rgba8_image_t
     > ImageTypes;
     typedef gil::any_image<ImageTypes> ImageType;
-
+#endif
     if (!fs::exists(path))
         throw BadFile("Texture file \"" + filename + "\" does not exist");
     if (!fs::is_regular_file(path))
@@ -274,7 +282,14 @@ void Texture::Load(const boost::filesystem::path& path, bool mipmap/* = false*/)
     m_default_height = Y(image.height());
     m_type = GL_UNSIGNED_BYTE;
 
-#if BOOST_VERSION >= 107000
+#if BOOST_VERSION >= 107400
+#define IF_IMAGE_TYPE_IS(image_prefix)                                  \
+    if (boost::variant2::get_if<image_prefix ## _image_t>(&image)) {    \
+        m_bytes_pp = sizeof(image_prefix ## _pixel_t);                  \
+        image_data = interleaved_view_get_raw_data(                     \
+            const_view(boost::variant2::get<image_prefix ## _image_t>(image))); \
+    }
+#elif BOOST_VERSION >= 107000
 #define IF_IMAGE_TYPE_IS(image_prefix)                                  \
     if (boost::get<image_prefix ## _image_t>(&image)) {                 \
         m_bytes_pp = sizeof(image_prefix ## _pixel_t);                  \

--- a/UI/GovernmentWnd.cpp
+++ b/UI/GovernmentWnd.cpp
@@ -537,6 +537,8 @@ void PoliciesListBox::HideAllCategories(bool refresh_list) {
         Populate();
 }
 
+using boost::placeholders::_1;
+using boost::placeholders::_2;
 
 //////////////////////////////////////////////////
 // GovernmentWnd::PolicyPalette                 //

--- a/UI/ObjectListWnd.cpp
+++ b/UI/ObjectListWnd.cpp
@@ -1886,9 +1886,9 @@ public:
         m_header_row->ColumnsChangedSignal.connect(
             boost::bind(&ObjectListBox::Refresh, this));
         m_header_row->ColumnHeaderLeftClickSignal.connect(
-            boost::bind(&ObjectListBox::SortingClicked, this, _1));
+            boost::bind(&ObjectListBox::SortingClicked, this, boost::placeholders::_1));
         m_obj_deleted_connection = GetUniverse().UniverseObjectDeleteSignal.connect(
-            boost::bind(&ObjectListBox::UniverseObjectDeleted, this, _1));
+            boost::bind(&ObjectListBox::UniverseObjectDeleted, this, boost::placeholders::_1));
     }
 
     virtual ~ObjectListBox()

